lib/smart_app_launch/token_response_body_test.rb

```diff
@@ -43,7 +43,7 @@ class TokenResponseBodyTest < Inferno::Test
       validate_required_fields_present(token_response_body, ['access_token', 'token_type', 'expires_in', 'scope'])
       validate_token_field_types(token_response_body)
       validate_token_type(token_response_body)
-      check_for_missing_scopes(requested_scopes, token_response_body)
+      check_for_missing_scopes(requested_scopes, token_response_body) unless config.options[:limited_launch]
```

Comment:
The meaning of `limited launch` is far from obvious. You should give this configuration option a name that says what it does.

Reply:
@Jammjammjamm I provided similar feedback (and more) but visibility may have been lost when the line of code changed?  I'm curious what you think here.

> You should update the README to reflect this new config option as you are essentially updating this test's "API": https://github.com/inferno-framework/smart-app-launch-test-kit#standalone-launch-group
> 
> I do wonder if it really makes sense for the default here to expect scopes requested equal scopes received, and to provide a warning if not (implying that servers SHOULD be granting scopes equal to those requested). Implying that servers are doing something wrong by allowing users to deny scopes is not correct, unless there are additional scenario/context-specific expectations being added here. Which I would argue are not appropriate for these base tests.
> 
> Seems like the default should be to provide no expectations. And then allow the test writers to layer on additional expectations in their own contexts. Either through a config (fail_if_any_scopes_denied; warn_if_any_scopes_denied), or just having the pattern be that you add another test that checks the scopes_requested / scopes_received inputs/outputs in your own context (which is what we are doing in g10, right? these warnings don't actually drive anything meaningful)?
> 
> I'll defer to Steve as he's architected these, but my vote would be to just downgrade the warnings to infos (at most) and not have these tests be concerned with context-specific expectations.

Reply:
While it's not wrong to allow users to deny scopes, it seems weird to me to deny scopes in a testing session unless you're explicitly testing the ability to deny scopes. I think a missing scope is most likely to be caused by some sort of configuration error and it makes sense to have them be very visible by default.

Reply:
We should have this soft expectation documented in the test description so the tester knows that they should approve all scopes if prompted (depending on this config flag).  Otherwise we are inventing a should requirement because that's definitely not coming from a spec (a system should grant all scopes -- no).

Reply:
@arscan added